### DISCUSSION TITLE
Custom x-axis intervall

### DISCUSSION THREAD

**Discussion by Sh1bbyz** [RomRider/apexcharts-card#458] [Q&A]
Hi,

What is the setting to adjust the x-axis intervall?
I have a three-day spanning graph but it will only show ether dates or first/last hour each day. I want it to show all hours or perhaps every third hour on the x-axis. 

![image](https://user-images.githubusercontent.com/114939764/208263573-f9260a12-6090-4b2f-85a1-0acaa5760d26.png)

`type: custom:apexcharts-card
header:
  show: true
  title: Förbrukning vs elpriser
  show_states: true
  colorize_states: true
now:
  show: true
  label: Nu
graph_span: 72h
span:
  start: day
  offset: '-24h'
apex_config:
  fill:
    type: gradient
    gradient:
      shadeIntensity: 0.1
      opacityFrom: 0.25
  xaxis:
    labels:
      format: HH
      show: true
      rotate: -45
      rotateAlways: true
      hideOverlappingLabels: false
      style:
        fontSize: 10
        fontWeight: 10
  chart:
    xaxis.type: datetime
    height: 300px
    zoom:
      enabled: false
    toolbar:
      show: false
      tools:
        zoom: false
        zoomin: false
        zoomout: false
        pan: false
        reset: true
  tooltip:
    fixed:
      enabled: true
      position: topLeft
yaxis:
  - id: second
    apex_config:
      min: 0
      max: auto
      opposite: true
      forceNiceScale: true
      decimalsInFloat: 2
      labels:
        show: true
  - id: first
    apex_config:
      min: 0
      max: auto
      opposite: false
      forceNiceScale: true
      decimalsInFloat: 0
      labels:
        show: true
all_series_config:
  stroke_width: 4
series:
  - entity: sensor.house_consumed_energy_from_grid
    yaxis_id: first
    name: Förbrukning / h
    type: area
    curve: stepline
    extend_to: now
    float_precision: 2
    stroke_width: 2
    opacity: 0.1
    color: orange
    show:
      in_header: true
      legend_value: false
    group_by:
      duration: 1h
      start_with_last: true
      func: diff
  - entity: sensor.nordpool_excl_vat
    yaxis_id: second
    name: Pris igår
    type: line
    curve: stepline
    extend_to: now
    float_precision: 3
    stroke_width: 3
    opacity: 1
    color: grey
    show:
      in_header: false
      legend_value: false
  - entity: sensor.nordpool_excl_vat
    yaxis_id: second
    name: Pris idag
    type: line
    curve: stepline
    extend_to: now
    float_precision: 3
    stroke_width: 3
    opacity: 1
    color: cyan
    show:
      in_header: true
      legend_value: false
    data_generator: |
      return entity.attributes.raw_today.map((p) => {
        return [new Date(p.start), p.value];
      });
  - entity: sensor.nordpool_excl_vat
    yaxis_id: second
    name: Pris imorgon
    type: line
    curve: stepline
    extend_to: now
    float_precision: 3
    stroke_width: 3
    opacity: 1
    color: magenta
    show:
      in_header: false
      legend_value: false
    data_generator: |
      return entity.attributes.raw_tomorrow.map((p) => {
        return [new Date(p.start), p.value];
      });`

**Comment by tarmor1** [comment]
Try going in directly with apex_config and set options there. Eg if I want to have one value for each hour and corresponding vertical lines too, I use:
```
apex_config:
  xaxis:
    tickAmount: 24
  grid:
    xaxis:
      lines:
        show: true
```
Check out the options directly from ApexCharts web: https://apexcharts.com/docs/

**Reply by Sh1bbyz** [reply]
Thanks @tarmor1 but I cannot get it to work. The grids can be shown with that but the tickamount doesnt do anything for the graph. The only thing is to reduce the graph to span for 48h, then it will be tickAmount 24 shown. However changing the tickAmount to anything else than 24 doesnt affect that graph either. Dunno if this is a bug or if I am missing something here. I am starting to believe in the former.

**Comment by tarmor1** [comment]
Can you share the full code maybe, along with Code formatting (Ctrl+E)

**Reply by Sh1bbyz** [reply]
@tarmor1 sure!

Dunno why the formatting wont work... here it is with formatting:
[https://pastebin.com/49kfMzbr](https://pastebin.com/49kfMzbr)

`type: custom:apexcharts-card
header:
  show: true
  title: Förbrukning vs elpriser
  show_states: true
  colorize_states: true
now:
  show: true
  label: Nu
graph_span: 72h
span:
  start: day
  offset: '-24h'
apex_config:
  fill:
    type: gradient
    gradient:
      shadeIntensity: 0.1
      opacityFrom: 0.25
  xaxis:
    tickAmount: 24
    grid:
      xaxis:
        lines:
          show: true
    labels:
      format: HH
      show: true
      rotate: -45
      rotateAlways: true
      hideOverlappingLabels: false
      style:
        fontSize: 10
        fontWeight: 10

  chart:
    xaxis.type: datetime
    height: 300px
    zoom:
      enabled: false
    toolbar:
      show: false
      tools:
        zoom: false
        zoomin: false
        zoomout: false
        pan: false
        reset: true
  tooltip:
    fixed:
      enabled: true
      position: topLeft
yaxis:
  - id: second
    apex_config:
      min: 0
      max: auto
      opposite: true
      forceNiceScale: true
      decimalsInFloat: 2
      labels:
        show: true
  - id: first
    apex_config:
      min: 0
      max: auto
      opposite: false
      forceNiceScale: true
      decimalsInFloat: 0
      labels:
        show: true
all_series_config:
  stroke_width: 4
series:
  - entity: sensor.energy_consumed
    yaxis_id: first
    name: Förbrukning / h
    type: area
    curve: stepline
    extend_to: now
    float_precision: 2
    stroke_width: 2
    opacity: 0.1
    color: orange
    show:
      in_header: true
      legend_value: false
    group_by:
      duration: 1h
      start_with_last: true
      func: diff
  - entity: sensor.nordpool
    yaxis_id: second
    name: Pris igår
    type: line
    curve: stepline
    extend_to: now
    float_precision: 3
    stroke_width: 3
    opacity: 1
    color: grey
    show:
      in_header: false
      legend_value: false
  - entity: sensor.nordpool
    yaxis_id: second
    name: Pris idag
    type: line
    curve: stepline
    extend_to: now
    float_precision: 3
    stroke_width: 3
    opacity: 1
    color: cyan
    show:
      in_header: true
      legend_value: false
    data_generator: |
      return entity.attributes.raw_today.map((p) => {
        return [new Date(p.start), p.value];
      });
  - entity: sensor.nordpool
    yaxis_id: second
    name: Pris imorgon
    type: line
    curve: stepline
    extend_to: now
    float_precision: 3
    stroke_width: 3
    opacity: 1
    color: magenta
    show:
      in_header: false
      legend_value: false
    data_generator: |
      return entity.attributes.raw_tomorrow.map((p) => {
        return [new Date(p.start), p.value];
      });`
